FABulous/fabric_generator/gds_generator/flows/fabric_macro_flow.py

```diff
@@ -0,0 +1,690 @@
+        fabric_width, fabric_height = self._compute_die_area(
+            row_heights,
+            column_widths,
+            halo_spacing,
+            tile_spacing,
+        )
+        info(f"Computed FABRIC_WIDTH (before rounding): {fabric_width}")
+        info(f"Computed FABRIC_HEIGHT (before rounding): {fabric_height}")
+
+        # Round the total fabric dimensions UP to the next pitch multiple
+        fabric_width_rounded = round_up_to_pitch(fabric_width, pitch_x)
+        fabric_height_rounded = round_up_to_pitch(fabric_height, pitch_y)
+
+        # Calculate the adjustment needed and distribute it to the halo
+        # Add the extra space to the right and top halo (keeps origin at 0,0)
+        width_adjustment = fabric_width_rounded - fabric_width
+        height_adjustment = fabric_height_rounded - fabric_height
+
+        halo_right = halo_right + width_adjustment
+        halo_top = halo_top + height_adjustment
+        halo_spacing = (halo_left, halo_bottom, halo_right, halo_top)
+
+        info(
+            f"Adjusted halo_spacing to: {halo_spacing} "
+            f"(width adj: {width_adjustment}, height adj: {height_adjustment})"
+        )
+        info(f"Final FABRIC_WIDTH: {fabric_width_rounded}")
+        info(f"Final FABRIC_HEIGHT: {fabric_height_rounded}")
+
+        self.config = self.config.copy(
+            DIE_AREA=[0, 0, fabric_width_rounded, fabric_height_rounded]
+        )
+
+        tile_spacing_x, tile_spacing_y = tile_spacing
+        tile_spacing_x = round_up_to_pitch(tile_spacing_x, pitch_x)
+        tile_spacing_y = round_up_to_pitch(tile_spacing_y, pitch_y)
+
+        # Place macros
+        cur_y = 0
+        for y, row in enumerate(reversed(self.fabric.tile)):
+            cur_x = 0
+            flipped_y = self.fabric.numberOfRows - 1 - y
+
+            for x, tile in enumerate(row):
+                tile_name = tile.name if tile is not None else None
+                prefix = f"Tile_X{x}Y{flipped_y}_"
+
+                for supertile_name, supertile in self.fabric.superTileDic.items():
+                    subtiles = [tile.name for tile in supertile.tiles]
+
+                    # Get the anchor of the supertile (bottom left)
+                    anchor = supertile.tileMap[-1][0]
+
+                    if tile_name in subtiles:
+                        if tile_name == anchor.name:
+                            tile_name = supertile_name
+
+                            # While the physical anchor is at the bottom left,
+                            # the anchor in FABulous is at the top left
+                            prefix = (
+                                f"Tile_X{x}Y{flipped_y - (len(supertile.tileMap) - 1)}_"
+                            )
+                        else:
+                            tile_name = None
+
+                if tile_name is None:
+                    info(f"Skipping {tile_name}")
+                else:
```

Comment:
```suggestion
                if tile_name is not None:
```

Wouldn't this make more sense? The original would always print `None` wouldn't it?